### PR TITLE
DSEC-1549: Add count for Azure SQL database

### PR DESCRIPTION
**Pull request by sam-orca**

### CODE REVIEW

**Review by antonkazlouski-orca — APPROVED** [review]

File: azure_resource_count.sh [hunk 1]
```diff
@@ -1,13 +1,14 @@
 #!/bin/bash
 
 LOG_FILE='azure_resource_count.log'
+DB_SIZE_THRESHOLD_IN_GB=1000
```
Comment (amine-orca) [inline azure_resource_count.sh:4]: Use 1024

File: azure_resource_count.sh [hunk 2]
```diff
@@ -238,6 +268,7 @@ echo "Serverless Containers Count: $ContainerCount (Workload Units: ${container_
 echo "Container Images Count: $containerImageCount (Workload Units: ${container_image_workloads})"
 echo "VM Images Count: $vmImageCount (Workload Units: ${vm_image_workloads})"
 echo "Container Hosts Count: $aksNodesCount (Workload Units: ${container_host_workloads})"
+echo "Managed DBs Hosts Count: (up to ${DB_SIZE_THRESHOLD_IN_GB} GB): ${db_host_workloads}"
```
Comment (amine-orca) [inline azure_resource_count.sh:271]: `Managed Databases` to be the same as AWS and GCP

File: azure_resource_count.sh [hunk 3]
```diff
@@ -190,7 +192,29 @@ for subscription in $subscriptions; do
...
+        # filter out db with size greater than 1TB
+        az sql db list-usages --subscription $subscription --server $server_name --resource-group $rg_name --name $db --query "[?currentValue.to_number(@) <= \`${db_size_threshold_in_bytes}\` && name=='database_allocated_size'].id" -o tsv > ${_temp_subscription_output} 2>> $LOG_FILE ||  echo "Failed to get Azure SQL Databases size for subscription ${subscription}"
+        # only one here
+        if [ -s ${_temp_subscription_output} ]; then
+          currentAzureDbCount=$((currentAzureDbCount + 1))
+        fi
+      done
+    done
+    if [ -n "$currentAzureDbCount" ]; then
+        echo "Managed DBs (up to ${DB_SIZE_THRESHOLD_IN_GB} GB): $currentAzureDbCount"
```
Comment (amine-orca) [inline azure_resource_count.sh:215]: `Managed Databases` instead

File: azure_resource_count.sh [hunk 4]
```diff
@@ -190,7 +192,29 @@ for subscription in $subscriptions; do
         echo "Container Hosts Count: $currentNodesCount"
         aksNodesCount=$((aksNodesCount + currentNodesCount))
     fi
-
+    # get azure sql databases
+    az sql server list --subscription $subscription --query "[].{name: name, resourceGroup: resourceGroup}" -o json> ${_temp_subscription_output} 2>> $LOG_FILE ||  echo "Failed to get Azure SQL Databases for subscription ${subscription}"
+    servers=$(cat "${_temp_subscription_output}")
+    currentAzureDbCount=0
+    db_size_threshold_in_bytes=$((DB_SIZE_THRESHOLD_IN_GB * 1000 * 1000 * 1000))
```
Comment (amine-orca) [inline azure_resource_count.sh:199]: Make `DB_SIZE_THRESHOLD_IN_GB` an overridable variable with a parameter like in AWS and GCP

**Pull request merged**